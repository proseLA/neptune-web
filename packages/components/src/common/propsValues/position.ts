export enum Position {
  BOTTOM = 'bottom',
  BOTTOM_LEFT = 'bottom-left',
  BOTTOM_RIGHT = 'bottom-right',
  CENTER = 'center',
  LEFT = 'left',
  LEFT_TOP = 'left-top',
  RIGHT = 'right',
  RIGHT_TOP = 'right-top',
  TOP = 'top',
}

export type PositionCenter = 'center';
export type PositionTop = 'top';
export type PositionRight = 'right';
export type PositionBottom = 'bottom';
export type PositionLeft = 'left';
