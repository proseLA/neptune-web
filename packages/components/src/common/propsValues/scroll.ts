export enum Scroll {
  CONTENT = 'content',
  VIEWPORT = 'viewport',
}

export type ScrollContent = 'content';
export type ScrollViewport = 'viewport';
