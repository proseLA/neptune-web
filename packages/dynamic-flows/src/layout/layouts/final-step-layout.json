[
  {
    "type": "heading",
    "text": "Details verified",
    "size": "lg",
    "margin": "lg",
    "align": "center"
  },
  {
    "type": "box",
    "width": "sm",
    "components": [
      {
        "type": "image",
        "url": "https://wise.com/public-resources/assets/spot-illustrations/dedicated_fraud_team.png",
        "text": "alt text",
        "margin": "lg"
      }
    ]
  },
  {
    "type": "paragraph",
    "text": "We do this extra check to make sure your money goes to the right place....",
    "align": "center"
  },
  {
    "type": "box",
    "width": "md",
    "components": [
      {
        "type": "button",
        "context": "success",
        "action": {
          "title": "Continue",
          "type": "primary",
          "exit": true,
          "data": { "someKey": "some value" },
          "url": "/success",
          "method": "GET"
        }
      }
    ]
  }
]
