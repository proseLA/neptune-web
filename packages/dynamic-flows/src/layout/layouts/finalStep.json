{
  "key": "account",
  "type": "final",
  "success": true,
  "details": {
    "image": "https://wise.com/public-resources/assets/spot-illustrations/dedicated_fraud_team.png",
    "title": "Details verified *",
    "description": "We do this extra check to make sure your money goes to the right place...."
  },
  "actions": [
    {
      "title": "Continue",
      "type": "primary",
      "exit": true,
      "data": { "someKey": "some value" },
      "url": "/success",
      "method": "GET"
    }
  ]
}
