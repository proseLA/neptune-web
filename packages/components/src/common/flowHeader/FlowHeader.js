import classNames from 'classnames';
import PropTypes from 'prop-types';
import { forwardRef } from 'react';

import { Layout } from '../propsValues/layouts';

const FlowHeader = forwardRef((props, reference) => {
  const { bottomContent, className, layout, leftContent, rightContent } = props;
  const isVertical = layout === Layout.VERTICAL;

  return (
    <div ref={reference} className={classNames('np-flow-header', 'd-flex', 'flex-wrap', className)}>
      <div
        className={classNames('align-items-center', 'np-flow-header__left', 'd-flex', {
          'flex__item--8': isVertical,
        })}
      >
        {leftContent}
      </div>

      <div
        className={classNames(
          'align-items-center',
          'd-flex',
          'np-flow-header__right',
          'justify-content-end',
          {
            'flex__item--4 ': isVertical,
            'order-2': !isVertical,
          },
        )}
      >
        {rightContent}
      </div>
      <div
        className={classNames('align-items-center', 'd-flex', 'justify-content-center', {
          'flex__item--12': isVertical,
          'order-1 flex-grow-1': !isVertical,
        })}
      >
        {bottomContent}
      </div>
    </div>
  );
});

FlowHeader.defaultProps = {
  bottomContent: undefined,
  className: undefined,
  layout: Layout.HORIZONTAL,
  leftContent: undefined,
  rightContent: undefined,
};

FlowHeader.propTypes = {
  bottomContent: PropTypes.node,
  className: PropTypes.string,
  layout: PropTypes.oneOf(['HORIZONTAL', 'VERTICAL']),
  leftContent: PropTypes.node,
  rightContent: PropTypes.node,
};

export default FlowHeader;
