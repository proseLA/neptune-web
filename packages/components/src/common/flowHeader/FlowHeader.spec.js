import '@testing-library/jest-dom';
import { render } from '@testing-library/react';

import { Layout } from '../propsValues/layouts';

import FlowHeader from './FlowHeader';

describe('FlowHeader', () => {
  const props = {
    leftContent: 'leftContent',
    rightContent: 'rightContent',
    bottomContent: 'bottomContent',
    className: 'className',
  };
  it('renders as expected', () => {
    expect(render(<FlowHeader {...props} />).container).toMatchSnapshot();
  });

  it('renders as expected with vertical layout', () => {
    expect(render(<FlowHeader {...props} layout={Layout.VERTICAL} />).container).toMatchSnapshot();
  });
});
