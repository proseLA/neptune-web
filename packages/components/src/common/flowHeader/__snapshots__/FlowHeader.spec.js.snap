// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlowHeader renders as expected 1`] = `
<div>
  <div
    class="np-flow-header d-flex flex-wrap className"
  >
    <div
      class="align-items-center np-flow-header__left d-flex"
    >
      leftContent
    </div>
    <div
      class="align-items-center d-flex np-flow-header__right justify-content-end order-2"
    >
      rightContent
    </div>
    <div
      class="align-items-center d-flex justify-content-center order-1 flex-grow-1"
    >
      bottomContent
    </div>
  </div>
</div>
`;

exports[`FlowHeader renders as expected with vertical layout 1`] = `
<div>
  <div
    class="np-flow-header d-flex flex-wrap className"
  >
    <div
      class="align-items-center np-flow-header__left d-flex flex__item--8"
    >
      leftContent
    </div>
    <div
      class="align-items-center d-flex np-flow-header__right justify-content-end flex__item--4 "
    >
      rightContent
    </div>
    <div
      class="align-items-center d-flex justify-content-center flex__item--12"
    >
      bottomContent
    </div>
  </div>
</div>
`;
